src/cli/rustup_mode.rs

```diff
@@ -792,12 +792,17 @@ async fn default_(
 }
 
 async fn check_updates(cfg: &Cfg<'_>, opts: CheckOpts) -> Result<utils::ExitCode> {
+    let t = cfg.process.stdout().terminal(cfg.process);
+    let is_a_tty = t.is_a_tty();
     let mut update_available = false;
     let channels = cfg.list_channels()?;
     let num_channels = channels.len();
     if num_channels > 0 {
-        let mp = MultiProgress::with_draw_target(ProgressDrawTarget::stdout());
-
+        let mp = if is_a_tty {
```

Comment:
Bonus: `cargo` uses the [`CARGO_TERM_PROGRESS_WHEN`](https://doc.rust-lang.org/cargo/reference/config.html#termprogresswhen) environment variable to control the drawing of the progress bar. I think it'd be interesting to mirror that here as well, maybe in a separate commit or PR.